webpack.hot.config.js

```diff
@@ -0,0 +1,23 @@
+/*
+ * Webpack development server configuration
+ */
+
+'use strict';
+
+var webpack = require('webpack');
+var baseConfig = require('./webpack.config.js');
+
+module.exports = Object.assign({}, baseConfig, {
+  debug: true,
+
+  entry: [
+    'webpack/hot/only-dev-server'
+  ].concat(baseConfig.entry),
+
+  plugins: baseConfig.plugins.concat([
+    new webpack.HotModuleReplacementPlugin(),
+    new webpack.NoErrorsPlugin()
+  ])
+});
+
+module.exports.module.loaders[0].loader = 'react-hot!babel-loader';
```

Comment:
Oh yes sorry the version that I tested worked because I saved the result in the `loaders` property instead of `module.loaders` and the baseConfig object was still updated by the method.

As you mentionned a `.concat` version of it would be fine like:

```js
module: {
  loaders: Array.prototype.concat([{
      test: /\.js$/,
      exclude: /node_modules/,
      loader: 'react-hot'
    }],
    baseConfig.module.loaders)
}
```